### PR TITLE
Fixed #27481 -- Made SQLite backend return annotated boolean values as boolean, not integer.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/27481

### CODE REVIEW

File: tests/annotations/tests.py [hunk 1]
```diff
@@ -488,3 +488,12 @@ class Lower(Func):
             ],
             lambda c: (c.name, c.tagline_lower)
         )
+
+    def test_boolean_value_annotation(self):
+        books = Book.objects.annotate(
+            is_book=Value(True, output_field=BooleanField()),
+            is_pony=Value(False, output_field=BooleanField()),
```
Comment (charettes) [inline tests/annotations/tests.py]: Might want to use `output_field=NullBooleanField()` here for larger coverage.

File: tests/annotations/tests.py [hunk 2]
```diff
@@ -488,3 +488,14 @@ class Lower(Func):
...
+            is_none=Value(None, output_field=NullBooleanField()),
+        )
+        for book in books:
+            self.assertIs(book.is_book, True)
+            self.assertIs(book.is_pony, False)
+            self.assertIs(book.is_none, None)
```
Comment (charettes) [inline tests/annotations/tests.py]: `self.assertIsNone` except for that the patch LGTM.

**Comment by timgraham** [comment]
I'm a bit nervous about assertions within loops in tests since if the loop is empty the assertions won't run. Do you think it's worth adding something like `self.assertGreater(len(books), 0)`?

**Comment by sir-sigurd** [comment]
@timgraham, makes sense. Are there any pitfalls with `self.assertTrue(books)`?

**Comment by timgraham** [comment]
It works but is perhaps less readable about the intent. I wouldn't veto it though.

**Pull request merged**